mathias_techera_project/globantProject/src/main/java/com/mathiastechera/project/cart/CartController.java

```diff
@@ -0,0 +1,81 @@
+package com.mathiastechera.project.cart;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.mathiastechera.project.user.User;
+
+@RestController
+@RequestMapping(value="/cart")
+public class CartController {
+	private final CartRepository repository;
+
+	CartController(CartRepository repository) {
+		this.repository = repository;
+	}
+	
+	/**
+	 * @return All the Items
+	 */
+	@GetMapping("/getAll")
+	List<Cart> allItems() {
+		return repository.findAll();		
+	}
+	
+	/**
+	 * Ask for a user to search his shopping cart.
+	 * TODO implement a token for different carts for the same user in different sessions.
+	 * @param user
+	 * @return a Cart object.
+	 */
+	@PostMapping("/show")
+	ResponseEntity<Cart> showCart(@RequestBody User user) {
+		if(user.getId() != null ) {
+			List<Cart> cartSearched = repository.findByBuyerID(user.getId());
+			if (cartSearched.size() == 1) {				
+				return ResponseEntity.ok(cartSearched.get(0));
+			}else if(cartSearched.size() == 0) {
+				return ResponseEntity.status(HttpStatus.NOT_FOUND).body(null);
+			}else {
+				return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(null);
+			}
+		}else {
+			return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(null);
+		}
+	}
+	
+
+	@PutMapping("/add")
+	ResponseEntity<Cart> addItem(@RequestBody CartItem cartitem) {
+		
+		if( cartitem.getBuyerID() != null && cartitem.getAmount() != null && cartitem.getItemID() != null && cartitem.getUnitValue() != null ) {			
```

Comment:
i would move this sort of validations to another place like a CarValidator class or something like that. There is another option which is relying in the repository since you should have said which fields are required and which aren't